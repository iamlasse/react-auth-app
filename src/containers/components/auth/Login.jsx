import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button, FormGroup, InputGroup, Intent } from '@blueprintjs/core';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import '@blueprintjs/core/lib/css/blueprint.css';
import '@blueprintjs/icons/lib/css/blueprint-icons.css';

const StyledFormGroup = styled(FormGroup)`
	display: flex;
	justify-content: space-around;
`;

const StyledInputGroup = styled(InputGroup)`
	margin-bottom: 1em;
`;

class Login extends Component {
	state = {
		disabled: false,
		inline: false,
		intent: Intent.NONE,
		username: '',
		password: ''
	};

	static propTypes = {
		login: PropTypes.func,
		fetching: PropTypes.bool
	};

	static defaultProps = {
		login: null,
		fetching: false
	};

	handleLogin = () => {
		const { login } = this.props;
		const { username, password } = this.state;
		login(username, password);
	};

	handleChange = e => {
		this.setState({ [e.target.id]: e.target.value });
	};

	render() {
		const { disabled, inline, intent, username, password } = this.state;
		const { fetching } = this.props;
		return (
			<form action="" style={{ maxWidth: 300, margin: 'auto' }}>
				<StyledFormGroup disabled={disabled} inline={inline} intent={intent}>
					<StyledInputGroup
						round
						leftIcon="person"
						large
						id="username"
						placeholder="Username"
						disabled={disabled}
						intent={intent}
						onChange={this.handleChange}
					/>
					<StyledInputGroup
						round
						large
						leftIcon="lock"
						type="password"
						id="password"
						placeholder="Password"
						disabled={disabled}
						intent={intent}
						onChange={this.handleChange}
					/>
					<div>
						<Button
							style={{ borderRadius: 50 }}
							disabled={!(username && password)}
							intent="primary"
							large
							fill
							loading={fetching}
							onClick={this.handleLogin}
						>
							Login
						</Button>
					</div>
					<span
						style={{
							display: 'flex',
							alignItems: 'center',
							marginTop: 20,
							justifyContent: 'space-between'
						}}
					>
						{"Don't have an account?"} <Link to="/register">Register</Link>
					</span>
				</StyledFormGroup>
			</form>
		);
	}
}

export default Login;
