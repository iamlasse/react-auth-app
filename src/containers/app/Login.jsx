import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button, FormGroup, InputGroup, Intent, Icon } from '@blueprintjs/core';
import styled from 'styled-components';
import { Link, withRouter } from 'react-router-dom';
import { actions as authActions } from '../../store/modules/auth';
import { bindActionCreators } from 'redux';
import '@blueprintjs/core/lib/css/blueprint.css';
import '@blueprintjs/icons/lib/css/blueprint-icons.css';

const StyledFormGroup = styled(FormGroup)`
	display: flex;
	justify-content: space-around;
`;

const StyledInputGroup = styled(InputGroup)`
	margin-bottom: 1em;
`;

class Login extends Component {
	state = {
		disabled: false,
		helperText: true,
		inline: false,
		intent: Intent.NONE,
		label: true,
		requiredLabel: false,
		username: '',
		password: ''
	};
	static propTypes = {};

	handleLogin = () => {
		console.log(this.state.username);
		this.props.login(this.state.username, this.state.password);
	};

	handleChange = e => {
		this.setState({ [e.target.id]: e.target.value });
	};
	render() {
		const { disabled, helperText, inline, intent, requiredLabel, label } = this.state;
		console.log(this.props);
		return (
			<form action="" style={{ maxWidth: 300, margin: 'auto' }}>
				<StyledFormGroup disabled={disabled} inline={inline} intent={intent}>
					<StyledInputGroup
						round
						leftIcon="person"
						large
						id="username"
						placeholder="Username"
						disabled={disabled}
						intent={intent}
						onChange={this.handleChange}
					/>
					<StyledInputGroup
						round
						large
						leftIcon="lock"
						type="password"
						id="password"
						placeholder="Password"
						disabled={disabled}
						intent={intent}
						onChange={this.handleChange}
					/>
					<div>
						<Button
							style={{ borderRadius: 50 }}
							disabled={!(this.state.username && this.state.password)}
							intent="primary"
							large
							fill
							loading={this.props.fetching}
							onClick={this.handleLogin}>
							Login
						</Button>
					</div>
					<span
						style={{
							display: 'flex',
							alignItems: 'center',
							marginTop: 20,
							justifyContent: 'space-between'
						}}>
						Don't have an account? <Link to="/register">Register</Link>
					</span>
				</StyledFormGroup>
			</form>
		);
	}
}

const mapStateToProps = state => ({
	fetching: state.auth.fetching,
	authenticated: state.auth.authenticated
});

const mapDispatchToProps = (dispatch, ownProps) => ({
	...bindActionCreators(
		{
			login: authActions.loginRequest
		},
		dispatch
	)
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login));
