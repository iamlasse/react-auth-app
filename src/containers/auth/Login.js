import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { actions as authActions, authSel } from '../../store/modules/auth';
import Login from '../components/auth/Login';

const { getAuthenticated, getAuthFetchStatus } = authSel;

const mapStateToProps = state => ({
	fetching: getAuthFetchStatus(state),
	authenticated: getAuthenticated(state)
});

const mapDispatchToProps = dispatch => ({
	...bindActionCreators(
		{
			login: authActions.loginRequest
		},
		dispatch
	)
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Login));
