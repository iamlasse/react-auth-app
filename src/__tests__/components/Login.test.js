import React from 'react';
import ReactDOM from 'react-dom';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { shallow, mount, render } from 'enzyme';
import Login from '../../containers/components/auth/Login';

let Component;
let wrapper;
describe('Login Component', () => {
	beforeEach(() => {
		wrapper = shallow(<Login />);
	});

	it('renders without crashing', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
