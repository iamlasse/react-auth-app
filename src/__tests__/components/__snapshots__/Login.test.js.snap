// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    fetching={false}
    login={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        action=""
        style={
          Object {
            "maxWidth": 300,
          }
        }
      >
        <h2
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Login
        </h2>
        <ForwardRef
          disabled={false}
          inline={false}
          intent="none"
        >
          <ForwardRef
            disabled={false}
            id="username"
            intent="none"
            large={true}
            leftIcon="person"
            onChange={[Function]}
            placeholder="Username"
            round={true}
          />
          <ForwardRef
            disabled={false}
            id="password"
            intent="none"
            large={true}
            leftIcon="lock"
            onChange={[Function]}
            placeholder="Password"
            round={true}
            type="password"
          />
          <div>
            <Blueprint3.Button
              disabled={true}
              fill="true"={true}
              intent="primary"
              large={true}
              loading={false}
              onClick={[Function]}
              style={
                Object {
                  "borderRadius": 50,
                }
              }
            >
              Login
            </Blueprint3.Button>
          </div>
          <span
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "space-between",
                "marginTop": 20,
              }
            }
          >
            Don't have an account?
             
            <Link
              replace={false}
              to="/register"
            >
              Register
            </Link>
          </span>
        </ForwardRef>
      </form>,
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#efefef",
        "display": "flex",
        "height": "100vh",
        "justifyContent": "center",
        "width": "100vw",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "",
        "children": Array [
          <h2
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Login
          </h2>,
          <ForwardRef
            disabled={false}
            inline={false}
            intent="none"
          >
            <ForwardRef
              disabled={false}
              id="username"
              intent="none"
              large={true}
              leftIcon="person"
              onChange={[Function]}
              placeholder="Username"
              round={true}
            />
            <ForwardRef
              disabled={false}
              id="password"
              intent="none"
              large={true}
              leftIcon="lock"
              onChange={[Function]}
              placeholder="Password"
              round={true}
              type="password"
            />
            <div>
              <Blueprint3.Button
                disabled={true}
                fill="true"={true}
                intent="primary"
                large={true}
                loading={false}
                onClick={[Function]}
                style={
                  Object {
                    "borderRadius": 50,
                  }
                }
              >
                Login
              </Blueprint3.Button>
            </div>
            <span
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "space-between",
                  "marginTop": 20,
                }
              }
            >
              Don't have an account?
               
              <Link
                replace={false}
                to="/register"
              >
                Register
              </Link>
            </span>
          </ForwardRef>,
        ],
        "style": Object {
          "maxWidth": 300,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
            "style": Object {
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": "Login",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                disabled={false}
                id="username"
                intent="none"
                large={true}
                leftIcon="person"
                onChange={[Function]}
                placeholder="Username"
                round={true}
              />,
              <ForwardRef
                disabled={false}
                id="password"
                intent="none"
                large={true}
                leftIcon="lock"
                onChange={[Function]}
                placeholder="Password"
                round={true}
                type="password"
              />,
              <div>
                <Blueprint3.Button
                  disabled={true}
                  fill="true"={true}
                  intent="primary"
                  large={true}
                  loading={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "borderRadius": 50,
                    }
                  }
                >
                  Login
                </Blueprint3.Button>
              </div>,
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "space-between",
                    "marginTop": 20,
                  }
                }
              >
                Don't have an account?
                 
                <Link
                  replace={false}
                  to="/register"
                >
                  Register
                </Link>
              </span>,
            ],
            "disabled": false,
            "inline": false,
            "intent": "none",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "id": "username",
                "intent": "none",
                "large": true,
                "leftIcon": "person",
                "onChange": [Function],
                "placeholder": "Username",
                "round": true,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
	margin-bottom: 1em;
",
                  ],
                },
                "displayName": "Styled(Blueprint3.InputGroup)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "id": "password",
                "intent": "none",
                "large": true,
                "leftIcon": "lock",
                "onChange": [Function],
                "placeholder": "Password",
                "round": true,
                "type": "password",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": true,
                  "rules": Array [
                    "
	margin-bottom: 1em;
",
                  ],
                },
                "displayName": "Styled(Blueprint3.InputGroup)",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": [Function],
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Blueprint3.Button
                  disabled={true}
                  fill="true"={true}
                  intent="primary"
                  large={true}
                  loading={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "borderRadius": 50,
                    }
                  }
                >
                  Login
                </Blueprint3.Button>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login",
                  "disabled": true,
                  "fill="true"": true,
                  "intent": "primary",
                  "large": true,
                  "loading": false,
                  "onClick": [Function],
                  "style": Object {
                    "borderRadius": 50,
                  },
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Don't have an account?",
                  " ",
                  <Link
                    replace={false}
                    to="/register"
                  >
                    Register
                  </Link>,
                ],
                "style": Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "space-between",
                  "marginTop": 20,
                },
              },
              "ref": null,
              "rendered": Array [
                "Don't have an account?",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Register",
                    "replace": false,
                    "to": "/register",
                  },
                  "ref": null,
                  "rendered": "Register",
                  "type": [Function],
                },
              ],
              "type": "span",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
	display: flex;
	justify-content: space-around;
",
              ],
            },
            "displayName": "Styled(Blueprint3.FormGroup)",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": [Function],
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          action=""
          style={
            Object {
              "maxWidth": 300,
            }
          }
        >
          <h2
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Login
          </h2>
          <ForwardRef
            disabled={false}
            inline={false}
            intent="none"
          >
            <ForwardRef
              disabled={false}
              id="username"
              intent="none"
              large={true}
              leftIcon="person"
              onChange={[Function]}
              placeholder="Username"
              round={true}
            />
            <ForwardRef
              disabled={false}
              id="password"
              intent="none"
              large={true}
              leftIcon="lock"
              onChange={[Function]}
              placeholder="Password"
              round={true}
              type="password"
            />
            <div>
              <Blueprint3.Button
                disabled={true}
                fill="true"={true}
                intent="primary"
                large={true}
                loading={false}
                onClick={[Function]}
                style={
                  Object {
                    "borderRadius": 50,
                  }
                }
              >
                Login
              </Blueprint3.Button>
            </div>
            <span
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "space-between",
                  "marginTop": 20,
                }
              }
            >
              Don't have an account?
               
              <Link
                replace={false}
                to="/register"
              >
                Register
              </Link>
            </span>
          </ForwardRef>
        </form>,
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#efefef",
          "display": "flex",
          "height": "100vh",
          "justifyContent": "center",
          "width": "100vw",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "action": "",
          "children": Array [
            <h2
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              Login
            </h2>,
            <ForwardRef
              disabled={false}
              inline={false}
              intent="none"
            >
              <ForwardRef
                disabled={false}
                id="username"
                intent="none"
                large={true}
                leftIcon="person"
                onChange={[Function]}
                placeholder="Username"
                round={true}
              />
              <ForwardRef
                disabled={false}
                id="password"
                intent="none"
                large={true}
                leftIcon="lock"
                onChange={[Function]}
                placeholder="Password"
                round={true}
                type="password"
              />
              <div>
                <Blueprint3.Button
                  disabled={true}
                  fill="true"={true}
                  intent="primary"
                  large={true}
                  loading={false}
                  onClick={[Function]}
                  style={
                    Object {
                      "borderRadius": 50,
                    }
                  }
                >
                  Login
                </Blueprint3.Button>
              </div>
              <span
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "space-between",
                    "marginTop": 20,
                  }
                }
              >
                Don't have an account?
                 
                <Link
                  replace={false}
                  to="/register"
                >
                  Register
                </Link>
              </span>
            </ForwardRef>,
          ],
          "style": Object {
            "maxWidth": 300,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "style": Object {
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "Login",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef
                  disabled={false}
                  id="username"
                  intent="none"
                  large={true}
                  leftIcon="person"
                  onChange={[Function]}
                  placeholder="Username"
                  round={true}
                />,
                <ForwardRef
                  disabled={false}
                  id="password"
                  intent="none"
                  large={true}
                  leftIcon="lock"
                  onChange={[Function]}
                  placeholder="Password"
                  round={true}
                  type="password"
                />,
                <div>
                  <Blueprint3.Button
                    disabled={true}
                    fill="true"={true}
                    intent="primary"
                    large={true}
                    loading={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "borderRadius": 50,
                      }
                    }
                  >
                    Login
                  </Blueprint3.Button>
                </div>,
                <span
                  style={
                    Object {
                      "alignItems": "center",
                      "display": "flex",
                      "justifyContent": "space-between",
                      "marginTop": 20,
                    }
                  }
                >
                  Don't have an account?
                   
                  <Link
                    replace={false}
                    to="/register"
                  >
                    Register
                  </Link>
                </span>,
              ],
              "disabled": false,
              "inline": false,
              "intent": "none",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "disabled": false,
                  "id": "username",
                  "intent": "none",
                  "large": true,
                  "leftIcon": "person",
                  "onChange": [Function],
                  "placeholder": "Username",
                  "round": true,
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "rules": Array [
                      "
	margin-bottom: 1em;
",
                    ],
                  },
                  "displayName": "Styled(Blueprint3.InputGroup)",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": [Function],
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "disabled": false,
                  "id": "password",
                  "intent": "none",
                  "large": true,
                  "leftIcon": "lock",
                  "onChange": [Function],
                  "placeholder": "Password",
                  "round": true,
                  "type": "password",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bwzfXH",
                    "isStatic": true,
                    "rules": Array [
                      "
	margin-bottom: 1em;
",
                    ],
                  },
                  "displayName": "Styled(Blueprint3.InputGroup)",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bwzfXH",
                  "target": [Function],
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Blueprint3.Button
                    disabled={true}
                    fill="true"={true}
                    intent="primary"
                    large={true}
                    loading={false}
                    onClick={[Function]}
                    style={
                      Object {
                        "borderRadius": 50,
                      }
                    }
                  >
                    Login
                  </Blueprint3.Button>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Login",
                    "disabled": true,
                    "fill="true"": true,
                    "intent": "primary",
                    "large": true,
                    "loading": false,
                    "onClick": [Function],
                    "style": Object {
                      "borderRadius": 50,
                    },
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Don't have an account?",
                    " ",
                    <Link
                      replace={false}
                      to="/register"
                    >
                      Register
                    </Link>,
                  ],
                  "style": Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "space-between",
                    "marginTop": 20,
                  },
                },
                "ref": null,
                "rendered": Array [
                  "Don't have an account?",
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Register",
                      "replace": false,
                      "to": "/register",
                    },
                    "ref": null,
                    "rendered": "Register",
                    "type": [Function],
                  },
                ],
                "type": "span",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": true,
                "rules": Array [
                  "
	display: flex;
	justify-content: space-around;
",
                ],
              },
              "displayName": "Styled(Blueprint3.FormGroup)",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": [Function],
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
